cmd/utils/nodecmd/chaincmd.go

```diff
@@ -232,45 +212,53 @@ func writeFailLog(key int, err error) {
 	logger.Crit(msg, "err", err)
 }
 
-func checkGenesisAndFillDefaultIfNeeded(genesis *blockchain.Genesis) *blockchain.Genesis {
-	engine := params.UseIstanbul
-	valueChanged := false
-	if genesis.Config == nil {
-		genesis.Config = new(params.ChainConfig)
+func ValidateGenesisConfig(g *blockchain.Genesis) error {
+	if g.Config.ChainID == nil {
+		return errors.New("chainID is not specified")
 	}
 
-	// using Clique as a consensus engine
-	if genesis.Config.Istanbul == nil && genesis.Config.Clique != nil {
-		engine = params.UseClique
-		if genesis.Config.Governance == nil {
-			genesis.Config.Governance = params.GetDefaultGovernanceConfig(engine)
-		}
-		valueChanged = true
-	} else if genesis.Config.Istanbul == nil && genesis.Config.Clique == nil {
-		engine = params.UseIstanbul
-		genesis.Config.Istanbul = params.GetDefaultIstanbulConfig()
-		valueChanged = true
-	} else if genesis.Config.Istanbul != nil && genesis.Config.Clique != nil {
-		// Error case. Both istanbul and Clique exists
-		logger.Crit("Both clique and istanbul configuration exists. Only one configuration can be applied. Exiting..")
+	if g.Config.Clique == nil && g.Config.Istanbul == nil {
+		return errors.New("consensus engine should be configured")
```

Comment:
It seems that the default engine is `UseIstanbul` if no engine is set. Do you have any reason to remove the default setting?